app/controllers/assignments_controller.rb

```diff
@@ -17,6 +17,8 @@ class AssignmentsController < ApplicationController
     :unaccept,
   ]
 
+  before_action :check_case_not_deleted, only: [:edit]
```

Comment:
I'm wondering about removing the `, only: [:edit]` here because there are other actions you shouldn't be able to do on deleted cases too - WDYT @mseedat-moj ?

Reply:
As discussed moved the check into `set_case_and_assignment` so that it gets applied to all these actions :)